### ISSUE TITLE
Resource: named queries

### ISSUE THREAD

**Issue by kvrhdn** [kvrhdn/terraform-provider-honeycombio#28]
A resource to manage named queries and their description.

More information: once you've built a query in the UI, you can name this query and add a description. This makes it easier to share this query. Additionally, this title is shown on boards.

API: *there is no API yet for this*

Possible Terraform schema:

```hcl
resource "honeycombio_query" "example" {
  name        = "My query"
  description = "A description of this query. _Supports Markdown._"

  dataset    = "my-dataset"
  query_json = data.honeycombio_query.my_query.json
}
```

To discuss:
- this might be confusing next to `data "honeycombio_query"`? Maybe rename the data source to `honeycombio_query_spec` so it's clear this is only the specification of query

cc. @fitzoh mentioned this in Slack

**Comment by kvrhdn** [comment]
This becomes possible with the query annotations API: https://docs-doug.honeycomb.io/api/query-annotations/

**Comment by kvrhdn** [comment]
This is now implemented with #63!
